File: parallel_internship_scraper_js/scraper/internshala.js
const axios = require("axios");
const cheerio = require("cheerio");

async function scrapeInternshala() {

  const url = "https://internshala.com/internships/software-development-internship/";

  const response = await axios.get(url);

  const $ = cheerio.load(response.data);

  const jobs = [];

  $(".individual_internship").each((i, el) => {

    const title =
      $(el).find(".job-title-href").text().trim() ||
      $(el).find("h3").text().trim();

    const company =
      $(el).find(".company-name").text().trim() ||
      $(el).find(".company_name").text().trim();

    if (title && company) {
      jobs.push({
        title: title,
        company: company,
        source: "Internshala"
      });
    }

  });

  return jobs;
}

module.exports = { scrapeInternshala };

File: 01_ai_internship_intelligence_engine/scraper/internshala.py
import requests
from bs4 import BeautifulSoup

def scrape_internshala():
    url = "https://internshala.com/internships/software-development-internship/"
    
    response = requests.get(url)
    soup = BeautifulSoup(response.text, "html.parser")

    jobs = []

    for card in soup.select(".individual_internship"):

        title = (
            card.select_one(".job-title-href") or
            card.select_one("h3")
        )

        company = (
            card.select_one(".company-name") or
            card.select_one(".company_name")
        )

        if title and company:
            jobs.append({
                "title": title.text.strip(),
                "company": company.text.strip(),
                "source": "Internshala"
            })

    return jobs